### PR TITLE
[TINY] Fix to #15763 - "collection selector was not NavigationExpansionExpression" when joining two FromSql()s

### PR DESCRIPTION
**Pull request by maumar**
Problem was that navigation expansion wasn't recognizing FromSqlOnQueryable as something that needs to be visited, so the expression was not being converted to NavigationExpansionExpression.
SelectMany requires collection navigation to be a NavigationExpansionExpression, hence the error.

### CODE REVIEW

File: src/EFCore/Query/NavigationExpansion/Visitors/NavigationExpandingVisitor_MethodCall.cs [hunk 1]
```diff
@@ -100,8 +100,8 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCallExp
                     return ProcessInclude(methodCallExpression);
 
                 //TODO: should we have relational version of this? - probably
-                case "FromSqlRaw":
-                    return ProcessFromRawSql(methodCallExpression);
+                case "FromSqlOnQueryable":
```
Comment (smitpatel) [inline src/EFCore/Query/NavigationExpansion/Visitors/NavigationExpandingVisitor_MethodCall.cs]: Can we do something better here? Like how EntityEquality works.
A method without any lambda (so no nav expansion needed) and source being same type as result type should be transparent. WIthTag can use the same processing.

Comment (maumar) [reply src/EFCore/Query/NavigationExpansion/Visitors/NavigationExpandingVisitor_MethodCall.cs]: i was thinking about it but wanted to do it in a future checkin. I guess I can incorporate it into this fix though.

**Comment by maumar** [comment]
new version up

File: src/EFCore/Query/NavigationExpansion/Visitors/NavigationExpandingVisitor_MethodCall.cs [hunk 2]
```diff
@@ -99,18 +95,48 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCallExp
...
+        }
+
+        private Expression ProcessUnknownMethod(MethodCallExpression methodCallExpression)
+        {
+            var resultSequenceType = TryGetNonPrimitiveSequenceType(methodCallExpression.Type);
+
+            // result is a sequence, no lambda arguments, exactly one generic argument corresponding to result sequence type
+            if (methodCallExpression.Object == null
+                && resultSequenceType != null
+                && methodCallExpression.Arguments.All(a => a.UnwrapQuote() == null)
```
Comment (smitpatel) [inline src/EFCore/Query/NavigationExpansion/Visitors/NavigationExpandingVisitor_MethodCall.cs]: Use a different function to identify if it is a lambda. UnwrapQuote needs to return LambdaExpression only, never null;

**Pull request merged**